komodo/yaml_file_types.py

```diff
@@ -38,16 +133,521 @@ class ManifestFile(YamlFile):
+    VALID_MAKES = ["rpm", "cmake", "sh", "pip", "rsync", "noop", "download"]
+
+    @staticmethod
+    def validate_package_name(package_name: str) -> bool:
+        if isinstance(package_name, str):
+            if str.islower(package_name):
+                return True
+            raise ValueError(f"Package name '{package_name}' should be lowercase.")
+        raise TypeError(f"Package name ({package_name}) should be of type string")
+
+    @staticmethod
+    def validate_package_version(package_name: str, package_version: str) -> bool:
+        if isinstance(package_version, str):
+            return True
+        raise TypeError(
+            f"Package '{package_name}' has invalid version type ({package_version})"
+        )
+
+    @staticmethod
+    def validate_package_entry(package_name: str, package_version) -> bool:
+        Package.validate_package_name(package_name)
+        Package.validate_package_version(package_name, package_version)
+
+    @staticmethod
+    def validate_package_entry_with_errors(
+        package_name: str, package_version: str
+    ) -> List[str]:
+        errors = []
+        try:
+            Package.validate_package_entry(package_name, package_version)
+        except (ValueError, TypeError) as e:
+            errors.append(str(e))
+        return errors
+
+    @staticmethod
+    def validate_package_importance(package_name: str, package_importance: str) -> bool:
+        if isinstance(package_importance, str):
+            if package_importance in Package.VALID_IMPORTANCES:
+                return True
+            raise ValueError(
+                f"{package_name} has invalid importance value ({package_importance})"
+            )
+        raise TypeError(
+            f"{package_name} has invalid importance type ({package_importance})"
+        )
+
+    @staticmethod
+    def validate_package_importance_with_errors(
+        package_name, package_importance: str
+    ) -> List[str]:
+        errors = []
+        try:
+            Package.validate_package_importance(package_name, package_importance)
+        except (ValueError, TypeError) as e:
+            errors.append(str(e))
+        return errors
+
+    @staticmethod
+    def validate_package_visibility(package_name: str, package_visibility: str) -> None:
+        if isinstance(package_visibility, str):
+            if package_visibility in Package.VALID_VISIBILITIES:
+                return True
+            raise ValueError(
+                f"Package '{package_name}' has invalid visibility value"
+                f" ({package_visibility})"
+            )
+        raise TypeError(
+            f"Package '{package_name}' has invalid visibility type"
+            f" ({package_visibility})"
+        )
+
+    @staticmethod
+    def validate_package_maturity(package_name: str, package_maturity: str) -> bool:
+        if isinstance(package_maturity, str):
+            if package_maturity in Package.VALID_MATURITIES:
+                return True
```

Comment:
It seems this function is only used in the function `validate_package_maturity_with_errors()`, and then that nothing ever relies on the truthiness of this function. Should this function just be inlined in `validate_package_maturity_with_errors()`?

Reply:
What is meant by truthiness of the function? I tried separating the responsibilities into two different functions, one checks the type and the value, and the other one collects the error(s) in the wanted format. It leads to extra code, but could be more reusable. Is it worth the trade off?

Reply:
I mean that I don't see any statements like `if validate_package_visibility(...):` which is a statement depending on whether that function returns True or False ("truthiness..")

Reply:
Ahhh, okay, I get it now. The return value is never used, so I will change it to return None instead.